src/System.IO.Compression.Brotli/System/IO/Compression/BrotliStream.cs

```diff
@@ -346,27 +292,20 @@ public override void Write(byte[] buffer, int offset, int count)
                     throw new TimeoutException(BrotliEx.TimeoutWrite);
                 }
                 copyLen = bytesRemain > _bufferSize ? _bufferSize : bytesRemain;
-                Marshal.Copy(buffer, currentOffset, _bufferInput, copyLen);
-                bytesRemain -= copyLen;
-                currentOffset += copyLen;
-                _availableInput = (nuint)copyLen;
-                _nextInput = _bufferInput;
-                while ((int)_availableInput > 0)
+                byte[] bufferInput = new byte[copyLen];
```

Comment:
Why is this extra array and copy needed? Why can't you use the input directly for the compress call?